app/components/QuoteForm.tsx

```diff
@@ -23,14 +37,54 @@ export default function QuoteForm({
   field2Placeholder,
   field3Placeholder,
   submitButtonText,
+  setSubmitSuccessful,
 }: QuoteFormProps) {
-  const handleSubmit = (event) => {
-    event.preventDefault();
-    // Form submission logic goes here
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+
+  const toast = useToast();
+
+  // EVENT HANDLERS
+  const onSubmit = (data: FormData) => {
+    sendEmail(data)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Network response was not ok.");
+        }
```

Comment:
This makes sense, thank you. I think I phrased my initial question poorly. My confusion was about line 56 and if it will result in going straight to the catch block on line 85 (if it didn't, there would be no feedback for the user about the error). As I thought more about it, and what I'm getting from your response, it will go into that catch block and the user will be notified of the error.